datamade/committee-oversight
# Add CommitteeRating model and import command

## Overview

Closes #77.

This PR:

- Defines a new `CommitteeRating` models. 
- Adds a `load_committeeratings` management command that loads in a hard-coded sample of ratings. 

## Notes

I chose to write a management command, rather than to include a fixture, because it's easier to update a sample dictionary than it is to update a fixture. Moreover, the management command can be repurposed to import structured ratings data when it becomes available.

## Questions

- I used a dump of the staging database to test the management command because `CommitteeRating` is related to `Committee`, but the dump included in this repository doesn't have any `Committee` objects. Should the dump be updated? Alternatively, I could update `import_data` to accept an argument instructing it to import committees (creating the `Committee` objects), but not hearings (which takes a long time). What do you think? **`Committee` is a vestigial model from work to sync up Lugar's spreadsheets with the scraped data. The ratings model should be associated with the OCD `Organization` model, which is in the dump.**
- Two questions about the new `Congress` model:
    - The new model would facilitate queries like `Congress.committeerating_set.all()` to get all ratings for a particular congress. Is this worth making queries more complicated? E.g., we'd want to select related objects when querying for ratings to avoid multiple DB calls. Happy to revert congress back to an attribute of the `CommitteeRating` model, if we only need to query by committee, not congress. **Factored this out. We don't want to query in this way.**
    - Open Civic Data [includes a legislative session model](https://github.com/opencivicdata/python-opencivicdata/blob/master/opencivicdata/legislative/models/session.py) that we could use instead of a custom `Congress` model. While this would allow us to lean yet further into the OCD standard, it requires start and end dates. Congressional start dates are inconsistent year over year, so we'd need to source and import these. This feels more like an enhancement to me, but I wanted to clear the choice to eschew this model – for now – with you!  **Tabling for now.**

## Testing Instructions

 * Stop your application containers and remove your Postgis volume: `docker-compose down && docker volume rm committee-oversight_committeeoversight-db-data`.
 * Restart your application containers and load in data as directed in the README. Apply the new migration: `docker-compose exec app python manage.py migrate`
 * Run the new management command: `docker-compose exec app python manage.py load_committeeratings`
 * Confirm the output looks like:
```
Added B rating for Senate Committee on Veterans' Affairs in 116th Congress
Added D rating for Senate Committee on Veterans' Affairs in 115th Congress
Added A rating for Senate Committee on Veterans' Affairs in 114th Congress
Added C rating for Senate Committee on Veterans' Affairs in 113th Congress
Added C- rating for Senate Committee on Veterans' Affairs in 112th Congress
Added A rating for Senate Committee on Veterans' Affairs in 111th Congress
Added C rating for Senate Committee on Veterans' Affairs in 110th Congress
Added B rating for Senate Committee on Veterans' Affairs in 109th Congress
Added F rating for Senate Committee on Veterans' Affairs in 108th Congress
Added F rating for Senate Committee on Veterans' Affairs in 107th Congress
Added C rating for Senate Committee on Energy and Natural Resources in 116th Congress
Added F rating for Senate Committee on Energy and Natural Resources in 115th Congress
Added C rating for Senate Committee on Energy and Natural Resources in 114th Congress
Added B+ rating for Senate Committee on Energy and Natural Resources in 113th Congress
Added D- rating for Senate Committee on Energy and Natural Resources in 112th Congress
Added A rating for Senate Committee on Energy and Natural Resources in 111th Congress
Added D rating for Senate Committee on Energy and Natural Resources in 110th Congress
Added B rating for Senate Committee on Energy and Natural Resources in 109th Congress
Added C rating for Senate Committee on Energy and Natural Resources in 108th Congress
Added F rating for Senate Committee on Energy and Natural Resources in 107th Congress
```
